server/src/main/java/org/elasticsearch/action/admin/cluster/settings/SettingsUpdater.java

```diff
@@ -94,8 +94,8 @@ synchronized ClusterState updateSettings(
             clusterSettings.validate(persistentFinalSettings, true);
 
             Metadata.Builder metadata = Metadata.builder(currentState.metadata())
-                .transientSettings(Settings.builder().put(transientFinalSettings).put(unknownOrInvalidTransientSettings).build())
-                .persistentSettings(Settings.builder().put(persistentFinalSettings).put(unknownOrInvalidPersistentSettings).build());
+                .transientSettings(transientFinalSettings.merge(unknownOrInvalidTransientSettings))
```

Comment:
At at least in my mind there's a bit of confusion with the new syntax if calling `merge` would update the 'this' object. It's an instance method, so most commonly in Java classes it would mean mutation of the original object, e.g sb.append.

It might be clearer if it was a static method, but I don't know if that complicates things. For example:

```
Settings.merge(transientFinalSettings, unknownOrInvalidTransientSettings)
```

It might even be possible to extend this to arbitrary length of Settings parameters with the ellipsis notation?